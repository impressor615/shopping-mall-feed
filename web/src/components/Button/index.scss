@import "~@/assets/styles/variables";

.btn {
  cursor: pointer;

  padding: 10px 15px;
  border: 1px solid $secondary;

  font-size: 16px;
  color: $secondary;

  opacity: 0.6;
  transition: all 0.3s ease-out;

  &:hover {
    color: $primary;
    font-weight: bold;

    background: $white;
    opacity: 1;
  }

  &.active {
    color: $primary;
    font-weight: bold;

    background: $white;
    opacity: 1;
  }
}
