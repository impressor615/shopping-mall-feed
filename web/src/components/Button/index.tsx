import "./index.scss";

import classnames from "classnames";
import React from "react";

interface ButtonProps {
  children: React.ReactNode;
  onClick?: ()  => void;
  active?: boolean;
  [rest: string]: any;
}

const Button = ({ children, onClick, active, ...rest }: ButtonProps) => (
  <button
    className={classnames("btn", { active })}
    onClick={onClick}
    {...rest}
  >
    { children }
  </button>
);

export default Button;
