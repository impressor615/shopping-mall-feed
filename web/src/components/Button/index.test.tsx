import enzyme from "enzyme";
import React from "react";

import Button from "@/components/Button";

describe("Button", () => {
  const children = "Button";
  it("should render children", () => {
    const wrapper = enzyme.shallow(<Button>{ children }</Button>);
    expect(wrapper.find(".btn").text()).toEqual(children);
  });

  it("should render active button when active prop is true", () => {
    const wrapper = enzyme.shallow(<Button active>{ children }</Button>);
    expect(wrapper.find(".btn.active")).toHaveLength(1);
  });

  it("should call onClick function when button is clicked", () => {
    const onClick = jest.fn();
    const wrapper = enzyme.shallow(<Button onClick={onClick}>{ children }</Button>);
    wrapper.find("button").simulate("click");
    expect(onClick).toHaveBeenCalled();
  });

  it("should be matched with snapshot", () => {
    const wrapper = enzyme.shallow(<Button>{ children }</Button>);
    expect(wrapper).toMatchSnapshot();
  });
});
