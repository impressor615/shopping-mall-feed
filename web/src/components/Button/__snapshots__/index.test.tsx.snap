// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be matched with snapshot 1`] = `
<button
  className="btn"
>
  Button
</button>
`;
